src/FSharpPlus/Extensions.fs

```diff
@@ -1,6 +1,8 @@
 namespace FSharpPlus
 
 open System
+open System.Collections.Generic
```

Comment:
This seems to be the reason why your PR is failing.
By opening this namespace at global level, `List` ( `ResizeArray` in F# jargon) shadows F# `List`.

Try opening it locally.